packages/markup-include/src/components/EditorPanel.tsx

```diff
@@ -0,0 +1,526 @@
+      )}
+    </>
+  )
+}
+
+type CheckBoxProps = {
+  fieldName: string
+  label: string
+  section?: string
+  subsection?: string
+  tooltip?: string
+  updateField: Function
+  value: boolean
+}
+
+const CheckBox: React.FC<CheckBoxProps> = memo(({ fieldName, label, subsection = null, tooltip, section = null, updateField, value, ...attributes }) => (
+  <label className='checkbox'>
+    <input
+      type='checkbox'
+      name={fieldName}
+      checked={value}
+      onChange={() => {
+        updateField(section, subsection, fieldName, !value)
+      }}
+      {...attributes}
+    />
+    <span className='edit-label column-heading'>{label}</span>
+    <span className='cove-icon'>{tooltip}</span>
+  </label>
+))
+type OpenControls = [Record<string, boolean>, Function] // useState type
+
+type EditorPanelProps = { children }
+
+const EditorPanel: React.FC<EditorPanelProps> = ({ children }) => {
+  const { config, data, isDashboard, loading, setParentConfig, updateConfig } = useContext(ConfigContext)
+  const { contentEditor, runtime, theme, visual, newViz } = config
+  const { inlineHTML, showHeader, srcUrl, title, useInlineHTML } = contentEditor
+  const [displayPanel, setDisplayPanel] = useState(true)
+  const updateField = updateFieldFactory(config, updateConfig, true)
+
+  const openVariableControls = useState({})
+
+  useEffect(() => {
+    // Pass up to Editor if needed
+    if (setParentConfig) {
+      const newConfig = convertStateToConfig()
+
+      setParentConfig(newConfig)
+    }
+  }, [config])
+
+  useEffect(() => {
+    let newConfig = { ...config }
+    delete newConfig.newViz
+    updateConfig(newConfig)
+  }, [])
+
+  const onBackClick = () => {
+    setDisplayPanel(!displayPanel)
+    updateConfig({
+      ...config,
+      showEditorPanel: !displayPanel
+    })
+  }
+
+  const Error = () => {
+    return (
+      <section className='waiting'>
+        <section className='waiting-container'>
+          <h3>Error With Configuration</h3>
+          <p>{runtime.editorErrorMessage}</p>
+        </section>
+      </section>
+    )
+  }
+
+  const Confirm = () => {
+    const confirmDone = e => {
+      e.preventDefault()
+      let newConfig = { ...config }
+      delete newConfig.newViz
+      updateConfig(newConfig)
+    }
+
+    return (
+      <section className='waiting'>
+        <section className='waiting-container'>
+          <h3>Finish Configuring</h3>
+          <p>Set all required options to the left and confirm below to display a preview of the markup.</p>
+          <button className='btn' style={{ margin: '1em auto' }} onClick={confirmDone}>
+            I'm Done
+          </button>
+        </section>
+      </section>
+    )
+  }
+
+  const convertStateToConfig = () => {
+    let strippedState = JSON.parse(JSON.stringify(config))
+    delete strippedState.newViz
+    delete strippedState.runtime
+
+    return strippedState
+  }
+
+  const [variableArray, setVariableArray] = useState<Variable[]>([])
```

Comment:
I thought we moved all this up to the parent config? are we sure this is getting saved?